boa/contracts/abi/abi_contract.py

```diff
@@ -34,7 +33,8 @@ def __init__(self, abi: dict, contract_name: str):
 
     @property
     def name(self) -> str:
-        return self._abi["name"]
+        # note: the `constructor` definition does not have a name
+        return self._abi.get("name") or self._abi["type"]
```

Comment:
what happens when we return `"constructor"` here? does anything rely on the value of name, or is it just for printing?

Reply:
probably returning `self._abi.get("name")` is more robust, caller has to handle the `None` case explicitly